airbyte-cdk/python/bin/remove-default-paginator-url-base.py

```diff
@@ -0,0 +1,76 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import os
+import re
+
+
+def get_all_manifest_paths(airbyte_integrations_path):
+    for connectors_path, connectors, files in os.walk(os.path.join(airbyte_integrations_path, "connectors")):
+        for connector in connectors:
+            if connector.startswith("source-"):
+                source_name = connector.replace("source-", "")
+                manifest_path = os.path.join(connectors_path, connector, f"source_{source_name}", f"{source_name}.yaml")
+                if os.path.isfile(manifest_path):
+                    yield manifest_path
+
+
+def find_default_paginators(manifest_entity):
+    if "type" in manifest_entity and manifest_entity["type"] == "DefaultPaginator":
+        return [manifest_entity]
+
+    default_paginators = []
+    for key, value in manifest_entity.items():
+        if isinstance(value, dict):
+            default_paginators.extend(find_default_paginators(value))
+    return default_paginators
+
+
+def has_default_paginator(manifest_lines):
+    return "DefaultPaginator" in "".join(manifest_lines)
+
+
+def remove_lines(lines):
+    """
+    This is a very imperfect implementation of remove DefaultPaginator.url_base. It is flawed because:
```

Comment:
This was my first attempt as well. From my local history:
```
if __name__ == "__main__":
    import argparse

    parser = argparse.ArgumentParser(description="Remove DefaultPaginator url_base")
    parser.add_argument("--path", help='airbyte-integrations folder')
    args = parser.parse_args()

    manifest_paths = get_all_manifest_paths(args.path)
    for manifest_path in manifest_paths:
        with open(manifest_path, "r") as manifest_file:
            manifest = yaml.safe_load(manifest_file.read())

        default_paginators = find_default_paginators(manifest)
        if default_paginators:
            for paginator in default_paginators:
                del paginator["url_base"]

            with open(manifest_path, 'w') as manifest_file:
                yaml.dump(manifest, manifest_file, default_flow_style=False)
```

However, the manifest would be rewritten in a different order. From the `git diff`:
```
--- a/airbyte-integrations/connectors/source-greenhouse/source_greenhouse/greenhouse.yaml
+++ b/airbyte-integrations/connectors/source-greenhouse/source_greenhouse/greenhouse.yaml
@@ -1,305 +1,303 @@
<...>
```

Since I don't control how the yaml library creates the dict, I have timeboxed the time spent on this solution

Reply:
Okay cool, I suppose that if any usages are missed then `./gradlew validateSourceYamlManifest` will show errors so your approach SGTM.